### PR TITLE
Show only public and pm channels in web chat

### PR DESCRIPTION
**Pull request by notbakaneko**
The other channel types apparently aren't a part of web chat.
- ~~`nonPmChannels` is now `publicChannels` because that's what it is (and another grouping is coming)~~
- channel list now renders channels according to their grouping (`GROUP` channels don't exist yet, so using default public channel sort)
- `channelList` moved to `ChatStateStore` because it's only used by that
- remove last usages of `inject`

### CODE REVIEW

File: resources/assets/lib/stores/channel-store.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import Message from 'models/chat/message';
 import core from 'osu-core-singleton';
 import UserStore from './user-store';
 
-const skippedChannelTypes = new Set<ChannelType>(['MULTIPLAYER', 'TEMPORARY']);
+const visibleChannelTypes = new Set<ChannelType>(['PM', 'PUBLIC']);
```
Comment (peppy) [inline resources/assets/lib/stores/channel-store.ts:21]: I don't think we use it yet, but `GROUP` type should probably also be displayed.

**Comment by nanaya** [comment]
Conflicts. And `publicChannels` now also includes `group` channels...? Maybe the previous `nonPmChannels` name is still better?

**Comment by nanaya** [comment]
conflicts :eyes:

**Comment by notbakaneko** [comment]
probably should make an alias for `Record<string, never>` at some point 👀

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**